Makefile

```
@@ -0,0 +1,264 @@
+reset := $(shell which tput > /dev/null && tput sgr0 2>/dev/null || echo "")
+bold  := $(shell which tput > /dev/null && tput bold 2>/dev/null || echo "")
+
+.PHONY: default
+
+default: generate
+
+help:
+	@echo "$(bold)Usage:$(reset) make $(cyan)<target>$(reset)"
+	@echo "  $(cyan)generate$(reset)                              - generate protocol buffers and plugin interface code"
+	@echo "  $(cyan)generate-check$(reset)                        - ensure generated code is up to date"
+	@echo "  $(cyan)protogen$(reset)                              - compile protocol buffers"
+	@echo "  $(cyan)protogen-check$(reset)                        - ensure generated protocol buffers are up to date"
+	@echo "  $(cyan)plugingen$(reset)                             - generate plugin interface code"
+	@echo "  $(cyan)plugingen-check$(reset)                       - ensure generated plugin interface code is up to date"
+	@echo
+	@echo "For verbose output set V=1"
+	@echo "  for example: $(cyan)make V=1 build$(reset)"
+
+# Used to force some rules to run every time
+FORCE: ;
+
+############################################################################
+# Service definitions and other protos
+############################################################################
+plugin-protos := \
+	private/proto/test/someplugin.proto \
+
+service-protos := \
+	private/proto/test/someservice.proto \
+	private/proto/test/somehostservice.proto \
+	proto/spire/service/common/config/v1/config.proto \
+
+grpc-protos := \
+	internal/proto/spire/service/private/init/v1/init.proto \
+
+protos := \
+	private/proto/test/echo.proto \
+
+
+############################################################################
+# OS/ARCH detection
+############################################################################
+os1=$(shell uname -s)
+os2=
+ifeq ($(os1),Darwin)
+os1=darwin
+os2=osx
+else ifeq ($(os1),Linux)
+os1=linux
+os2=linux
+else
+$(error unsupported OS: $(os1))
+endif
+
+arch1=$(shell uname -m)
+ifeq ($(arch1),x86_64)
+arch2=amd64
+else ifeq ($(arch1),aarch64)
+arch2=arm64
+else
+$(error unsupported ARCH: $(arch1))
+endif
+
+############################################################################
+# Vars
+############################################################################
+
+build_dir := $(DIR)/.build/$(os1)-$(arch1)
+
+go_version_full := $(shell cat .go-version)
+go_version := $(go_version_full:.0=)
+go_dir := $(build_dir)/go/$(go_version)
+go_bin_dir := $(go_dir)/bin
+go_url = https://storage.googleapis.com/golang/go$(go_version).$(os1)-$(arch2).tar.gz
+go_path := PATH="$(go_bin_dir):$(PATH)"
+
+golangci_lint_version = v1.27.0
+golangci_lint_dir = $(build_dir)/golangci_lint/$(golangci_lint_version)
+golangci_lint_bin = $(golangci_lint_dir)/golangci-lint
+
+protoc_version = 3.14.0
+ifeq ($(arch1),aarch64)
+protoc_url = https://github.com/protocolbuffers/protobuf/releases/download/v$(protoc_version)/protoc-$(protoc_version)-$(os2)-aarch_64.zip
+else
+endif
```

It seems that this else should not be empty
```suggestion
protoc_url = https://github.com/protocolbuffers/protobuf/releases/download/v$(protoc_version)/protoc-$(protoc_version)-$(os2)-$(arch1).zip
endif
```

---

Wow, I don't know how that happened :laughing: